src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java

```diff
@@ -370,6 +382,48 @@ public void validate(MonitorableJob.Status status) {
         }
     }
 
+    public void validateMobility(MonitorableJob.Status status) {
+
+        // Sometimes this method is called when no status object is available.
+        if (status == null) status = new MonitorableJob.Status();
+
+        // VALIDATE GTFS feed
+        FileReader fr = null;
+        try {
+            LOG.info("Beginning MobilityData validation...");
+            status.update("MobilityData Analysis...", 11);
+
+            File gtfsZip = this.retrieveGtfsFile();
+            // Namespace based folders avoid clash for validation being run on multiple versions of a feed
+            // TODO: do we know that there will always be a namespace?
+            String validatorOutputDirectory = "/tmp/datatools_gtfs/" + this.namespace + "/";
+
+            // Set up MobilityData validator
+            ValidationRunnerConfig.Builder builder = ValidationRunnerConfig.builder();
+            builder.setGtfsSource(gtfsZip.toURI());
+            builder.setOutputDirectory(Path.of(validatorOutputDirectory));
+            ValidationRunnerConfig mbValidatorConfig = builder.build();
+
+            // Run MobilityData validator
+            ValidationRunner runner = new ValidationRunner(new VersionResolver());
+            runner.run(mbValidatorConfig);
+
+            // Read generated report and save to Mongo
+            fr = new FileReader(validatorOutputDirectory + "report.json");
```

Comment:
Change to try with resources:

`String json;`
`try (FileReader fr = new FileReader(validatorOutputDirectory + "report.json")) {`
`    BufferedReader in = new BufferedReader(fr);`
`    json = in.lines().collect(Collectors.joining(System.lineSeparator()));`
`}`